### DISCUSSION TITLE
Per-file M4A ratings?

### DISCUSSION THREAD

**Discussion by cdwfs** [quodlibet/quodlibet#3876] [Q&A]
Hello! I'm trying to confirm my understanding of QL's support for per-file ratings tags in M4A files (ratings as the equivalent of ID3's POPM field and/or QL's 0-5 star scale, not the "content rating" atom). I gather there are conflicting conventions for how ratings are stored in M4A -- some applications use a 'rate' atom, others, store the rating in some sort of XTRA tag, and [mp3tag](https://www.mp3tag.de/) offers several different "Extended Tag" options for ratings (WinAmp, WMP, MM). As far as I can tell, Quod Libet ignores all of these tags; if I set them through an external tool, songs still appear in QL with the default rating. Furthermore, it seems that per-song ratings set within QL do not get saved to the audio file at all; ratings on M4A files will persist if QL is quit and relaunched (which implies they're stored _somewhere_, maybe in library-level metadata?), but if an M4A file is given a rating, removed from the library, and re-added, it will return to the default rating.

Is this the expected behavior? If so, are there any plans to support at least one of the per-file ratings conventions for M4A?

**Comment by declension** [comment]
You need to enable saving ratings to the file in the prefs (and then it's under an email prefix).

As for M4A, support is a bit limited yes. see #3342 #2932 #323 

Perhaps we could convert this to a feature request for one of those conventions for ratings in M4A...
